src/scan-options.ts

```diff
@@ -25,15 +54,54 @@ export interface ScanOptions {
 export type KeyTypeForScanOptions<O extends ScanOptions> = O extends {
   indexName: string;
 }
-  ? [secondary: string, primary: string]
+  ? ScanOptionIndexedStartKey
   : string;
 
-export interface ScanOptionsRpc {
+/**
+ * When providing a start key for an indexed scan you need to provide the
+ * secondary index to scan at. You may also provide a primary index.
+ */
+export type ScanOptionIndexedStartKey =
+  | [secondary: string, primary: string]
+  | [secondary: string];
```

Comment:
We could also allow a `string` which would have the same semantics as `[secondary: string]`. WDYT?

Reply:
One reason to do this is the potential footgun if not using TypeScript and you end passing `{key: 'abc'}`... Without allowing `string` that would have the same semantics as `{key: ['a', 'b']}`

Reply:
I like the idea of allowing a bare string for the index case as a convenience.